R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+    maintaining = str_detect(count_as_disposed, "(?i)\\bmaint.*(place|dwelling)"),
+    tax = str_detect(count_as_disposed, "(?i)\\btax\\b"),
+    minor = str_detect(count_as_disposed, "(?i)minor|child|under.*21"),
+    inmate = str_detect(count_as_disposed, "(?i)inmate|jail|contrab|penal|\\bfac"),
+    vehicle = str_detect(count_as_disposed, "(?i)(?<!\\snon|not\\s)(?:\\sMV\\b|motor vehicle)"),
+    proceeds = str_detect(count_as_disposed, "(?i)ac.*\\bp(r|o){0,2}|proceed"),
+    larceny = str_detect(count_as_disposed, "(?i)larceny|theft|LMFR|larceny of merchandise|meat|corporeal property|\\bstol(e|en)\\b"),
+    receive_stolen = str_detect(count_as_disposed, "(?i)RCSP|stolen property|stoeln|concealing stolen"),
+    embezzle = str_detect(count_as_disposed, "(?i)embezzlement|embezlement|emblezzlement|embzzlement"),
+    misc = str_detect(count_as_disposed, "(?i)fish|domesticated game|oil|drilling|gas"),
+    pawn = str_detect(count_as_disposed, "(?i)repay pawn|pawnbroker|pawn shop|in pawn"),
+    hospitality = str_detect(count_as_disposed, "(?i)(hotel|inn|restaurant|boarding house|rooming house|motel|auto camp|trailer camp|apartment|rental unit|rental house)"),
+    forgery = str_detect(count_as_disposed, "(?i)forgery|forged|counterfeit|bogus check"), # Andrew: added "bogus check"
+    drug = str_detect(count_as_disposed, "(?i)CDS|C\\.D\\.S|DRUG|OXY|HUFF|AMPHET|ZOLOL|ZOLAM|HYDROC|CODEIN|PRECURS|XANAX|MORPH|METERDI|ZEPAM|LORAZ|VALIU|EPHED|SUB|COCA|PSEUDO| CS|CS | CD|CD |\\bPRESCRIP|\\bNARC|\\bMETH|\\bC\\.D\\.|HEROIN|ANHYD|AMMONIA|OPIUM|LORTAB|\\bPARAPH\\b|\\bMA.*NA\\b|\\bMJ\\b|\\bMARI\\b"),
+    impersonate_officer = str_detect(count_as_disposed, "(?i)personating police"), # Andrew: added
+    tax_stamp = str_detect(count_as_disposed, "(?i)tax stamp"), # Andrew: added
+    a_b_upon = str_detect(count_as_disposed, "(?i)a b "), # This was getting picked up by UCCS property
+    uccs_drug = probability >= 0.8 & uccs_code >= 3090 & uccs_code <= 3162,
+    uccs_drug_plus = probability >= 0.8 & uccs_code >= 3090 & uccs_code <= 3230,
+    uccs_property = probability >= 0.8 & uccs_code %in% property_codes$uccs_code,
+    # All regex disqualifying a charge from SQ780; this will make the below chunks more readable.
+    disqualifying_regex = if_else(
+      dist | intent | paraphernalia | weapon | trafficking | endeavor | wildlife |
+        manufacture | commercial | school_park_church | conspiracy | maintaining |
+        minor | inmate | vehicle | proceeds | impersonate_officer | tax_stamp |
+        a_b_upon,
+      TRUE, FALSE
+    ),
+    # Any drug regex? Not worried about uccs codes with this variable yet
+    regex_drugs = if_else(poss & drug, TRUE, FALSE),
+    # Any property crime regex? Not worried about uccs codes with this variable
+    regex_props = if_else(larceny | receive_stolen | embezzle | misc | pawn | hospitality | forgery, TRUE, FALSE)
+  ) |>
+  arrange(count_as_disposed)
```

Comment:
I made some changes to this chunk that might be useful to know going forward:

- You only need to write `(?i)` once at the beginning of your regex strings; it basically "turns on" case insensitive mode, so you don't have to repeat it again afterward for items separated by a `|`. Also, if you want to turn case insensitive mode back off later on in a string, you can do it with `(?-i)`
- I added new regex variables `impersonate_officer`, `tax_stamp`, and `a_b_upon`; these were just tweaks to catch things I saw slipping through (mostly just because of UCCS false positives).
- I added new variables `disqualifying_regex`, `regex_drugs`, and `regex_props` -- these kind of sum up the other regex variables (e.g. did it have any regex related to sq780 property crimes? anything from the disqualifying regexes like intent to distribute or tax stamp?) so that the next chunk will be easier to read.

Reply:
`a_b_upon` is for "A B UPON AN OFFICER" type charges, which were getting classified by UCCS as property crimes for some reason.